pkg/cmd/describe/describe_rollout.go

```diff
@@ -0,0 +1,533 @@
+		}
+
+		// Calculate ready status
+		readyContainers := 0
+		for _, containerStatus := range pod.Status.ContainerStatuses {
+			if containerStatus.Ready {
+				readyContainers++
+			}
+		}
+		podInfo.Ready = fmt.Sprintf("%d/%d", readyContainers, len(pod.Spec.Containers))
+
+		// Calculate revision
+		if pod.Labels["pod-template-hash"] != "" {
+			podInfo.Revision = pod.Labels["pod-template-hash"]
+		} else if pod.Labels["controller-revision-hash"] != "" {
+			podInfo.Revision = pod.Labels["controller-revision-hash"]
+		}
+
+		workloadInfo.Pod = append(workloadInfo.Pod, podInfo)
+		return nil
+	})
+
+	if err != nil {
+		return nil, err
+	}
+
+	return workloadInfo, nil
+}
+
+func (o *DescribeRolloutOptions) colorizeIcon(phase string) string {
+	if o.NoColor || phase == "" {
+		return ""
+	}
+	switch phase {
+	case string(rolloutsapi.RolloutPhaseHealthy):
+		return "\033[32m✔\033[0m"
+	case string(rolloutsapi.RolloutPhaseProgressing):
+		return "\033[33m⚠\033[0m"
+	case string(rolloutsapi.RolloutPhaseDisabled), string(rolloutsapi.RolloutPhaseTerminating), string(rolloutsapi.RolloutPhaseDisabling):
+		return "\033[31m✘\033[0m"
+	case string(rolloutsapi.RolloutPhaseInitial):
+		return "\033[33m⚠\033[0m"
+	default:
+		return ""
+	}
+}
+
+func (o *DescribeRolloutOptions) printRolloutInfo(rollout *rolloutsapi.Rollout) {
+	// Name, Namespace, Status, Message
+	fmt.Fprintf(o.Out, tableFormat, "Name:", rollout.Name)
+	fmt.Fprintf(o.Out, tableFormat, "Namespace:", rollout.Namespace)
+	if rollout.Status.ObservedGeneration == rollout.GetObjectMeta().GetGeneration() {
+		fmt.Fprintf(o.Out, tableFormat, "Status:", o.colorizeIcon(string(rollout.Status.Phase))+" "+string(rollout.Status.Phase))
+
+		if rollout.Status.Message != "" {
+			fmt.Fprintf(o.Out, tableFormat, "Message:", rollout.Status.Message)
+		}
+	}
+
+	// Strategy
+	fmt.Fprintf(o.Out, tableFormat, "Strategy:", "Canary")
+
+	if canary := rollout.Spec.Strategy.Canary; canary != nil {
+		fmt.Fprintf(o.Out, tableFormat, " Step:", strconv.Itoa(len(canary.Steps))+"/"+strconv.Itoa(int(rollout.Status.CanaryStatus.CurrentStepIndex)))
+		fmt.Fprint(o.Out, " Steps:\n")
+		currentStepIndex := int(rollout.Status.CanaryStatus.CurrentStepIndex)
+		for i, step := range canary.Steps {
+			isCurrentStep := (i + 1) == currentStepIndex
+			if isCurrentStep {
+				fmt.Fprint(o.Out, "\033[32m")
+			}
+
+			if step.Replicas != nil {
+				fmt.Fprintf(o.Out, tableFormat, "  -  Replicas: ", step.Replicas)
+			}
+			if step.Traffic != nil {
+				fmt.Fprintf(o.Out, tableFormat, "     Traffic: ", *step.Traffic)
+			}
+
+			if step.Matches != nil {
+				fmt.Fprintf(o.Out, tableFormat, "     Matches: ", step.Matches)
+			}
+
+			if isCurrentStep {
+				fmt.Fprint(o.Out, "\033[0m") // Reset color
+			}
+		}
+
+		// Traffic Routings
+		if len(rollout.Spec.Strategy.Canary.TrafficRoutings) > 0 {
```

Comment:
for end-to-end canary release, it is also possible for rollout to use rollout.spec.strategy.canary.trafficRoutingRef to specify traffic routing obj, plz refer to 
https://openkruise.io/rollouts/user-manuals/strategy-end2end-canary-update